cmd/kubeadm/app/componentconfigs/controllermanager.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package componentconfigs
+
+import (
+	"strconv"
+
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+)
+
+const (
+	// defaultNodeMaskCIDRIPv4 is default mask size for IPv4 node cidr for use by the controller manager
+	defaultNodeMaskCIDRIPv4 = 24
+
+	// DefaultNodeMaskCIDRIPv6 is default mask size for IPv6 node cidr for use by the controller manager
+	defaultNodeMaskCIDRIPv6 = 64
+)
+
+// GetClusterNodeMask return the corresponding node-cidr-mask
+// based on the Cluster configuration and the IP family
+// Default is 24 for IPv4 and 64 for IPv6
+func GetClusterNodeMask(cfg *kubeadmapi.ClusterConfiguration, isIPv6, isDualStack bool) int {
```

Comment:
can we add GetClusterNodeMask in validation.go and lowercase it given it is used only there?

ideally we should not have this file cmd/kubeadm/app/componentconfigs/controllermanager.go, because the componentconfigs package is reserved for kubelet and kube-proxy for the time being.

Reply:
Yeah, I was wondering what will be the best place, I choose the wrong one :)